### ISSUE TITLE
triplet loss can't convergence,how can I do?

### ISSUE THREAD

**Issue by sjtutsb** [xiaolonw/caffe-video_triplet#8]
hi,
   I use your code to train triplet loss completely, but,I find the loss value can't convergence, it show that loss value=margin all the time when I train face image with your model,why? expect your reply,thank you very much!

**Comment by HAHA-DL** [comment]
@DreamHorse0301 Hi, can I ask how you deal with this problem finally?

**Comment by poirotZ** [comment]
@sjtutsb @HAHA-DL  HI, I also have this problem, how you deal with it?

**Comment by AhmedAbdel-Aal** [comment]
I came across this issue, I don't know if you still need an answer but for future readers:

The loss=margin, mainly means that all the embeddings collapsed to one point. Hence, the similarity function is giving zero for your loss with no differentiating between positive and negative pairs, leaving the margin only the non-zero term in the loss equation.

a general tip would be to decrease the learning rate so the networks are more prone to collapse.
